torch/jit/__init__.py

```diff
@@ -1870,6 +1870,9 @@ def check_unique(param):
             if buf is not None:
                 tmp_module._buffers[name] = buf
                 check_unique(buf)
+        for name, val in orig.__dict__.items():
+            if torch._C._jit_is_traceable_ivalue(val) and name not in orig._parameters and name not in orig._buffers:
```

Comment:
This condition is a little opaque without context, can we:
1. add a comment explaining what the rule here is, and 2
2. Add `orig._modules` so its clear that we're saying "python attributes that not part of the `nn.Module` state"

Reply:
we should also check_unique here I guess